### DISCUSSION TITLE
WinUI3 - How to add an external plugin dynamically

### DISCUSSION THREAD

**Discussion by annmanvel** [microsoft/WindowsAppSDK#2073] [Dev team discussions]
I am doing a POC to load a third-party plugin to our main application at runtime, without any project reference or .dll reference.
What are the steps we need to achieve this?

**Comment by BenJKuhn** [comment]
This is a pretty open-ended question. Is there more detail you can provide? Without knowing anything about the plug-in it's hard to provide useful information.

Ben

**Reply by Entangled-Logic** [reply]
Same question. I want to register a WinUI component in the registry and add an identifier to an XML project file so the component can be late bound and loaded dynamically, hiding the implementation behind interfaces defined in a proxy-stub server. Without having to add the reference to the component in the Package.appxmanifest and adding a switch and headers for each component.
